add_basemap: keep original axis limits of figure

jorisvandenbossche opened the pull request:
Opening this for discussion: in my original implementation (http://geopandas.readthedocs.io/en/latest/gallery/plotting_basemap_background.html#contextily-helper-function) I reset the axis limits to the original ones when the `ax` is passed to the function. 

One reason to do this: the extent of the figure does not depend on the zoom level (on how many tiles are merged).

Closes https://github.com/darribas/contextily/issues/45

darribas commented:
Nice! so, just to make sure I get it correctly, this PR will make sure the extent of the plot will stay that of the original plot, not that of the pulled down full image, right?

jorisvandenbossche commented:
(will add a test if you approve the change)

darribas commented:
Yes, I think this makes sense. Quick question, would this also apply to the case where `url` is a local file?

jorisvandenbossche commented:
Updated the PR to fix merge conflicts.

> Yes, I think this makes sense. Quick question, would this also apply to the case where url is a local file?

For now, I didn't make any distinction between both.

And I also updated the test

darribas commented:
Cool. One more thing, could you please add the option of `reset_extent` as mentioned in #45? I think you're right it makes more sense to set the default to `True`, but it I think in some contexts (particularly when `url` points to a local file), the user might be interested in letting the extent be reset by the addition.

darribas commented on contextily/plotting.py:
```diff
@@ -92,8 +93,12 @@ def add_basemap(ax, zoom=ZOOM, url=sources.ST_TERRAIN,
     ax.imshow(image, extent=extent, 
               interpolation=interpolation, **extra_imshow_args)
+
+    ax.axis((xmin, xmax, ymin, ymax))
```
I think giving the option should just be adding here:

```python
if reset_extent:
     ax.axis(...)
```

darribas commented:
@jorisvandenbossche I've cleaned conflicts in one of the tests with current `master` and added the option to reset the extent (as opposed of *always* doing it) into a [PR](https://github.com/jorisvandenbossche/contextily/pull/1) to your branch. If you merge that one, I'm happy to merge this too into `master`.

darribas commented:
Whoops, looks like this introduces errors on the tests now...

https://travis-ci.org/darribas/contextily/jobs/522345969

These are in addition to the 3.5 issue and come up in 3.6. Would you mind having a look into it @jorisvandenbossche?
